### PR TITLE
Xy 6

### PR DESCRIPTION
**Pull request by Jiaqi-beep**
Original NCL Script: https://www.ncl.ucar.edu/Applications/Scripts/xy_6.ncl
Original NCL Plot: https://www.ncl.ucar.edu/Applications/Images/xy_6_lg.png
Issue: #330 

xy6

### CODE REVIEW

**Comment by Jiaqi-beep** [comment]
The plot isn't exactly the same. I think it's because of how the xticks are assigned. I'm not exactly sure how the original plot creates equal spacing on the tickmarks on x-axis while the actual values(years) are not equally spaced apart.

**Comment by michaelavs** [comment]
> The plot isn't exactly the same. I think it's because of how the xticks are assigned. I'm not exactly sure how the original plot creates equal spacing on the tickmarks on x-axis while the actual values(years) are not equally spaced apart.

So I think I have made a (very) hacky work around for this issue with ```ax.xaxis.set_minor_locator``` and also defining the xticks in line 72 and removing the null ('') labels from xticklabels list. Like I said, it's hacky, so @anissa111 if you know of a better route than this, definitely let me know but, either way, here is what I have inserted on line 77:
```
ax.xaxis.set_minor_locator(FixedLocator([1954, 1961, 1968.5, 1977, 1987]))
```
this will put exactly one tick in middle point of each major tick at the specified value. You can use decimals, so it can be precise enough. But what I don't love about this version is that it specified by hand and I have a feeling it may not be the true blue best approach. 

Let me know what you think!

**Comment by Jiaqi-beep** [comment]
> > The plot isn't exactly the same. I think it's because of how the xticks are assigned. I'm not exactly sure how the original plot creates equal spacing on the tickmarks on x-axis while the actual values(years) are not equally spaced apart.
> 
> So I think I have made a (very) hacky work around for this issue with `ax.xaxis.set_minor_locator` and also defining the xticks in line 72 and removing the null ('') labels from xticklabels list. Like I said, it's hacky, so @anissa111 if you know of a better route than this, definitely let me know but, either way, here is what I have inserted on line 77:
> 
> ```
> ax.xaxis.set_minor_locator(FixedLocator([1954, 1961, 1968.5, 1977, 1987]))
> ```
> 
> this will put exactly one tick in middle point of each major tick at the specified value. You can use decimals, so it can be precise enough. But what I don't love about this version is that it specified by hand and I have a feeling it may not be the true blue best approach.
> 
> Let me know what you think!

I tried to follow what you said but I don't seem to be able to

**Comment by Jiaqi-beep** [comment]
Oops sorry, I don't know what I did that closed the PR.

**Comment by Jiaqi-beep** [comment]
> > The plot isn't exactly the same. I think it's because of how the xticks are assigned. I'm not exactly sure how the original plot creates equal spacing on the tickmarks on x-axis while the actual values(years) are not equally spaced apart.
> 
> So I think I have made a (very) hacky work around for this issue with `ax.xaxis.set_minor_locator` and also defining the xticks in line 72 and removing the null ('') labels from xticklabels list. Like I said, it's hacky, so @anissa111 if you know of a better route than this, definitely let me know but, either way, here is what I have inserted on line 77:
> 
> ```
> ax.xaxis.set_minor_locator(FixedLocator([1954, 1961, 1968.5, 1977, 1987]))
> ```
> 
> this will put exactly one tick in middle point of each major tick at the specified value. You can use decimals, so it can be precise enough. But what I don't love about this version is that it specified by hand and I have a feeling it may not be the true blue best approach.
> 
> Let me know what you think!

I tried to edit my code according to this but I still couldn't recreate the original plot. The plot itself doesn't seem to change for me when I change the ticks locations. Do you mind sharing your code from line 72-77? Thank you!

**Comment by michaelavs** [comment]
@Jiaqi-beep 

The code starts on line 66 and goes through 77:
```
# Use geocat.viz.util convenience function to set axes parameters without calling several matplotlib functions
# Set axes limits, tick values, and tick labels
gvutil.set_axes_limits_and_ticks(ax,
                                 xlim=(1951, 1991),
                                 ylim=(-1.5, 0.6),
                                 yticks=np.arange(-1.2, .6, .4),
                                 xticks=[1951, 1957, 1965, 1972, 1982, 1991],
                                 xticklabels=[
                                     '1951', '1957', '1965', '1972',
                                     '1982', '1991'
                                 ])
ax.xaxis.set_minor_locator(FixedLocator([1954, 1961, 1968.5, 1977, 1987]))
```

I will note that the new minor tick marks are pretty faint, but that might be something that can be changed with a kwarg later on.

This is the plot output that I have when running locally:
image

**Comment by Jiaqi-beep** [comment]
@michaelavs Your version totally makes sense to me! That's why I was confused how so it seems that the x-axis tick marks in the original plot seem to space out evenly, which was what I tried to mimic. Am I correct in thinking that we should modify original ncl plot when needed? 
Also, when @anissa111 and I talked briefly the other day, we both noticed that the bend on the plot (between 1957-65) in our version points toward the right, but it points toward the left in the original plot. Are we supposed to fix little differences like these?

**Comment by anissa111** [comment]
I'm not concerned about the slight difference in that one point. Also, I like the idea of moving the years around to make physical sense, even if that's not happening in the original NCL. I have no idea what the original plot was doing with regards to the labels and spacing.

**Comment by Jiaqi-beep** [comment]
updated.
Figure_1

File: Plots/XY/NCL_xy_6.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/xy_6_lg.png
+"""
+
+###############################################################################
+# Import packages:
+
+import numpy as np
+import xarray as xr
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FixedLocator
```
Comment (anissa111) [inline Plots/XY/NCL_xy_6.py:18]:
```suggestion
```

Unused import

**Review by anissa111 — COMMENTED** [review]
This looks good to me! I There's an unused input, but I'll approve this after that!

**Review by michaelavs — APPROVED** [review]
This looks good to me! I changed the location of the tick mark between 1982 and 1991 to 1986.5 instead of 1987 to be the true halfway point, but I didn't think making you do the full change on your end was worth it before approving on my side!

**Review by anissa111 — APPROVED** [review]
Looks good to me, too!

**Pull request merged**